### PR TITLE
XSS sniff: require escaping for die(), exit(), printf()

### PR DESCRIPTION
**Pull request by JDGrimes**
Other functions we should possibly add: https://stackoverflow.com/questions/4673747/what-php-functions-create-output

See #170

### CODE REVIEW

File: WordPress/Tests/XSS/EscapeOutputUnitTest.inc [hunk 1]
```diff
@@ -60,4 +60,12 @@ echo $foo ? $foo : 'no foo'; // Bad
...
\ No newline at end of file
+echo 4; // OK
+
+exit( $foo ); // Bad
+exit( esc_html( $foo ) ); // OK
+
+die( $foo ); // Bad
+die( esc_html( $foo ) ); // OK
+
+printf( 'Hello %s', $foo ); // Bad
```
Comment (westonruter) [inline WordPress/Tests/XSS/EscapeOutputUnitTest.inc]: @JDGrimes I think `printf` here is a tricky one, and it is discussed in #212. We basically need recognize that what you have here is indeed bad, but whitelist the following as OK:

```php
printf( 'Hello %s', esc_html( $foo ) ); // OK
```

That will require some sophistication, so I think this that `printf` and `vprintf` should be removed from the scope of this PR, unless you want to tackle #212 as well :smile:

Comment (JDGrimes) [inline WordPress/Tests/XSS/EscapeOutputUnitTest.inc]: I've just pushed some improved tests in 39881e8 that demonstrate that this works for `printf()`. It also addresses `vprintf()`. `sprintf()` is still another story, which I/someone will tackle at a later date. :-)

**Comment by westonruter** [comment]
Impressive!

**Comment by corvannoorloos** [comment]
I believe this gives an error on:
```php
if ( ! defined( 'ABSPATH' ) ) {
	exit;
}
```
As I probably don't comprehend the complete change, would you suggest also escaping this?

**Comment by JDGrimes** [comment]
@corvannoorloos That is a bug. I'll create a PR to fix it shortly.

**Comment by corvannoorloos** [comment]
@JDGrimes thanks!

**Pull request merged**